src/applications/static-pages/ebenefits-link-update.js

```diff
@@ -0,0 +1,30 @@
+import { isLoggedIn as isLoggedInSelector } from 'platform/user/selectors';
+import { shouldUseProxyUrl } from 'platform/site-wide/ebenefits/selectors';
+import { proxyUrl } from 'platform/site-wide/ebenefits/utilities';
+
+export default function updateEbenefitsLinks(store) {
+  let unsubscribe;
+  const storeListener = () => {
+    const state = store.getState();
+    const isLoggedIn = isLoggedInSelector(state);
+    const useProxyUrl = shouldUseProxyUrl(state);
+
+    if (isLoggedIn && useProxyUrl) {
+      const ebenefitsDomain = 'https://www.ebenefits.va.gov';
+      const ebenefitsLinks = [
+        ...document.querySelectorAll(`a[href^="${ebenefitsDomain}"]`),
+      ];
+
+      ebenefitsLinks.forEach(anchor => {
+        const { href } = anchor;
+        const path = href.replace(ebenefitsDomain, '');
```

Comment:
This will either mess up React apps or be overwritten by them on render. Why aren't we just always sending users through the proxy regardless of their signed in status?

Reply:
It's part of static-pages so it only renders on static content/Drupal pages

Reply:
For example, go to https://www.va.gov/health-care/eligibility/ and search for `Request a representative` -> link to eBen that is in a WYSIWYG

Reply:
@rianfowler unfortunately the way the proxy presently works, if the user doesn't have an active SSOe session, it's going to just redirect them to the home page. I don't believe that logic can be changed, so the burden of checking authenticated state falls on us

Reply:
I think this redirect behavior should be done on a backend service. Scattering this all over the site is really brittle, hard to change, and hard to test. `vets-api` has the user state and can make this determination and then we would only need to link to that in one way on the site.